packages/aragon-api/src/index.js

```diff
@@ -346,7 +346,15 @@ export class AppProxy {
       map(v => v || {}))
     const latestBlock$ = this.web3Eth('getBlockNumber')
     // init the app state with the cached state
-    const initState$ = init ? cacheValue$.pipe(switchMap(({ state }) => from(init(state)))) : from([null])
+    const initState$ = init
+      ? cacheValue$.pipe(
+        switchMap(({ state }) => from(init(state))),
+        delayWhen((initState) => {
+          console.debug('- store - init state:', initState)
+          return this.cache('state', initState)
```

Comment:
This I assume ensures two things:
- The initialised state is persistent to cache
- The `forkJoin` will only take the value once `this.cache` is successful. 

Am I correct?

Reply:
Yes on both.

Unfortunately there's still a bit of a gap with the initialized state being flushed to `state` and the frontend rendering, but this will be taken care of (and we can remove this bit) once we [stop using the cache for `state`](https://github.com/aragon/aragon.js/pull/313#issuecomment-496577875)